source/extensions/tracers/opentelemetry/span_context.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace OpenTelemetry {
+
+/**
+ * This class represents the context of an OpenTelemetry span, including the following
+ * characteristics: trace id, span id, parent id, and trace flags.
+ */
+class SpanContext {
+public:
+  /*
+   * Default constructor creates an empty context.
+   */
+  SpanContext() = default;
+
+  /*
+   * Constructor that creates a context object from the supplied attributes.
+   */
+  SpanContext(const absl::string_view& version, const absl::string_view& trace_id,
+              const absl::string_view& parent_id, bool sampled)
+      : version_(version), trace_id_(trace_id), parent_id_(parent_id), sampled_(sampled) {}
+
+  /**
+   * @return the span's version as a hex string.
+   */
+  std::string version() const { return version_; }
```

Comment:
I think you can return `const std::string&` to avoid copying. Other accessors as well.

Reply:
Nice, good call! Updated.